### PR TITLE
Improve support for nvcc_wrapper with exotic host compiler

### PR DESCRIPTION
**Pull request by masterleinad**
Related to https://github.com/kokkos/kokkos/issues/3125#issuecomment-657699072 fixing two issues:
- `PGI` doesn't understand the usual warning flags (and we deal with that correctly when `PGI` is the compiler) but if using `nvcc_wrapper` and `PGI` as host compiler we still try to use the default flags. Since we don't have any special flags for `nvcc` right now, it makes sense to pass flags according to `CMAKE_CXX_COMPILER` instead of `KOKKOS_CXX_COMPILER`.
- `PGI` uses different standard flags (like `--c++11` instead of `-std=c++11`) than `NVCC` and this is true all compilers except for `g++` and `clang++` AFAICT (https://github.com/Kitware/CMake/search?p=1&q=CMAKE_CXX11_STANDARD_COMPILE_OPTION&unscoped_q=CMAKE_CXX11_STANDARD_COMPILE_OPTION). Hence, it only makes sense to test for compatibility if the host compiler is `g++` or `clang++`. It seems that we only try to pass a standard flag to `nvcc` through `nvcc_wrapper` in this case and not for, e.g. `PGI` but that's maybe for another pull request.

### CODE REVIEW

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Retest this please.

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -98,9 +98,10 @@ IF(KOKKOS_ENABLE_COMPILER_WARNINGS)
     ${COMMON_WARNINGS})
 
   COMPILER_SPECIFIC_FLAGS(
-    PGI NO-VALUE-SPECIFIED
-    GNU     ${GNU_WARNINGS}
-    DEFAULT ${COMMON_WARNINGS}
+    COMPILER_ID CMAKE_CXX_COMPILER
```
Comment (jjwilke) [inline cmake/kokkos_arch.cmake:101]: Slightly confused by this. Shouldn't this be `CMAKE_CXX_COMPILER_ID` ?

**Review by jjwilke — CHANGES_REQUESTED** [review]

File: cmake/kokkos_arch.cmake (hunk 1 above)
Comment (masterleinad) [reply cmake/kokkos_arch.cmake]: Thanks, fixed!

**Review by jjwilke — APPROVED** [review]

**Pull request merged**